steps/pipeline_tests.yaml

```
@@ -0,0 +1,43 @@
+parameters:
+- name: environment
+  default: ''
+- name: domain_prefix
+  default: ''
+- name: playbook_file
+  default: 'pipeline-tests.yaml'
+- name: workingDirectory
+  default: ''
+- name: junit_output_dir
+  default: ''
+- name: root_domain
+  default: platform.hmcts.net
+
+steps:
+- task: Bash@3
+  displayName: Run pipeline tests
+  inputs:
+    targetType: 'inline'
+    script: |
+      mkdir junit
+      echo "localhost ansible_connection=local" > /tmp/hosts
+      ansible-playbook ${{ parameters.playbook_file }} -i /tmp/hosts
+    workingDirectory: '${{ parameters.workingDirectory }}'
+    failOnStderr: true
+  env:
+    ANSIBLE_LOCALHOST_WARNING: false
+    ANSIBLE_CALLBACKS_ENABLED: junit
+    JUNIT_FAIL_ON_IGNORE: true
+    JUNIT_OUTPUT_DIR: '${{ parameters.junit_output_dir }}'
+    ${{ if eq(parameters.environment, 'prod') }}:
+      GLOBAL_DOMAIN_NAME: '${{ parameters.root_domain }}'
+    ${{ else }}:
+      GLOBAL_DOMAIN_NAME: '${{ parameters.domain_prefix }}.${{ parameters.root_domain }}'
+
+- task: PublishTestResults@2
+  displayName: Publish pipeline tests
+  condition: always()
+  inputs:
+    testResultsFormat: 'JUnit'
+    failTaskOnFailedTests: true
+    testResultsFiles: '${{ parameters.junit_output_dir }}/pipeline-tests-*.*.xml'
+    testRunTitle: 'Platform Pipeline Tests (${{ parameters.environment }})'
```

this is probably enough
```suggestion
    testRunTitle: 'Tests (${{ parameters.environment }})'
```